vision_agent/tools/tool_utils.py

```diff
@@ -34,61 +35,58 @@ def send_inference_request(
     files: Optional[List[Tuple[Any, ...]]] = None,
     v2: bool = False,
     metadata_payload: Optional[Dict[str, Any]] = None,
-) -> Dict[str, Any]:
+) -> Any:
     # TODO: runtime_tag and function_name should be metadata_payload and now included
     # in the service payload
-    try:
```

Comment:
Sure

Reply:
there is a third way to call the public app (task based instead of tool based), I just created another function to address that and separated a common function, which is 1. create trace 2. call post 3. record url trace